examples/src/main/java/tachyon/examples/JournalCrashTest.java

```diff
@@ -0,0 +1,377 @@
+   * Class to record the correct status, which are the success operations of each client thread.
+   * The status will be stored in a local temp file and also load from it.
+   */
+  static class StatusRecord {
+    /**
+     * Read the status from a local file.
+     * @param localFile the path of the local file
+     * @return the loaded StatusRecord
+     * @throws ClassNotFoundException
+     * @throws IOException
+     */
+    public static StatusRecord readStatus(String localFile)
+        throws ClassNotFoundException, IOException {
+      StatusRecord status = new StatusRecord();
+      ObjectInputStream obIn = new ObjectInputStream(new FileInputStream(localFile));
+      status.setStatus((List<ClientOpType>) obIn.readObject(), (List<Integer>) obIn.readObject(),
+          (List<String>) obIn.readObject());
+      obIn.close();
+      return status;
+    }
+
+    private List<ClientOpType> mOpTypes = null;
+    private List<Integer> mSuccessNums = null;
+    private List<String> mWorkDirs = null;
+
+    public List<ClientOpType> getOpTypes() {
+      return mOpTypes;
+    }
+
+    public List<Integer> getSuccessNums() {
+      return mSuccessNums;
+    }
+
+    public List<String> getWorkDirs() {
+      return mWorkDirs;
+    }
+
+    public void record(List<ClientThread> clientThreads) {
+      mOpTypes = new ArrayList<ClientOpType>(clientThreads.size());
+      mSuccessNums = new ArrayList<Integer>(clientThreads.size());
+      mWorkDirs = new ArrayList<String>(clientThreads.size());
+      for (ClientThread client : clientThreads) {
+        mOpTypes.add(client.getOpType());
+        mSuccessNums.add(client.getSuccessNum());
+        mWorkDirs.add(client.getWorkDir());
+      }
+    }
+
+    /**
+     * Set the status values. This is only used when reading status from local file.
+     */
+    private void setStatus(List<ClientOpType> opTypes, List<Integer> successNums,
+        List<String> workDirs) {
+      mOpTypes = opTypes;
+      mSuccessNums = successNums;
+      mWorkDirs = workDirs;
+    }
+
+    /**
+     * Write this status to a local file.
+     * @param localFile the path of the local file
+     * @throws IOException
+     */
+    public void writeStatus(String localFile) throws IOException {
+      ObjectOutputStream obOut = new ObjectOutputStream(new FileOutputStream(localFile));
+      obOut.writeObject(mOpTypes);
+      obOut.writeObject(mSuccessNums);
+      obOut.writeObject(mWorkDirs);
+      obOut.close();
+    }
+  }
+
+  // The two Exit Codes are used to tell script if the test runs well.
+  private static final int EXIT_FAILED = 99;
+  private static final int EXIT_SUCCESS = 100;
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private static int sClientNum = 10;
+  private static TachyonURI sMasterAddress = null;
+  private static String sTestDir = null;
+  private static List<Thread> sClientThreadList = null;
+  private static List<ClientThread> sClientThreads = null;
```

Comment:
One is used to wait all the threads finished (`thread.join()`). Another is used to get the result (`clientThread.getSuccessNum()`).

I can use `ClientThread extends Thread` to only have one list.